### ISSUE TITLE
HTML cleanup for ActivityPub breaks lists and quotes

### ISSUE THREAD

**Issue by kvibber** [pfefferle/wordpress-activitypub#285]
Currently, the plugin appears to be stripping all formatting except for links and paragraphs, which matches Mastodon's current ability to display notes. Some fediverse platforms do support limited formatting, though, including some Mastodon forks, Misskey and its forks, GoToSocial, and others.

1. Bold and Italics: It would be helpful to preserve these tags for platforms that accept them. Vanilla Mastodon will just strip out the tags and it'll look the same as it does now.

2. Blockquote:  Stripping blockquotes or converting them to paragraphs *can break the post's meaning*, and preserving them for platforms that support it keeps post meaning intact. It doesn't fix the problem for Mastodon, but there's a discussion over there on what formatting to include, so it's possible future versions may recognize them.

3. Lists: Stripping out all the list item tags is breaking the format completely, turning lists into long lines of sentence fragments without even spaces between them. Fewer platforms seem to support them in my testing (ex: GoToSocial does, Misskey doesn't), but there are better ways to handle them than leaving everything on one line. My inclination here would be to at least insert a line break between list items, and maybe a number or asterisk to mimic the way people type lists in plaintext.

**Comment by pfefferle** [comment]
Is this too much whitelisting? https://github.com/WordPress/WordPress/blob/06305406dfad83d90f864202aa0dedb16e558438/wp-includes/kses.php#L68

**Comment by kvibber** [comment]
I'm not sure. I've been trying out various platforms to see how they interact with each other and reporting bugs (like the one with this and GoToSocial a week or so ago - thanks for the quick turnaround on it!), and the ones I've tried so far all look like they should be able to handle the tags and just sanitize away the ones *they* don't want.

I looked up [Mastodon's sanitizing scheme](https://docs.joinmastodon.org/spec/activitypub/) and they have a list of known block elements that they convert to paragraphs, and they specifically replace <li> elements with a <br /> at the end of the line. And I [ran a test with an unordered list](https://gts.keysmash.xyz/@kelsonv/statuses/01GTB7KP9B38AEYQ5HAGJFYQ7J) from GoToSocial, and confirmed that Takahe and Calckey do the same.

I've got test accounts on several ActivityPub platforms, a couple of my own test servers, and a test WordPress blog separate from my main. I'd be happy to test any changes on them.
